tests/test_ols.py

```diff
@@ -714,16 +720,65 @@ def test_rolling_least_squares(window_size: int, min_periods: int, use_woodbury:
             window=window_size,
             min_nobs=min_periods,
             expanding=True,
+            missing="drop",
         ).fit()
+        coef_statsmodels = (
+            pl.DataFrame(mdl.params, schema=["x1", "x2"])
+            .fill_nan(None)
+            .with_columns(pl.all().forward_fill())
+        )
+
     assert np.allclose(
-        coef_rolling,
-        mdl.params,
+        coef_rolling.to_numpy(),
+        coef_statsmodels.to_numpy(),
         rtol=1.0e-3,
         atol=1.0e-3,
         equal_nan=True,
     )
 
 
+@pytest.mark.parametrize("window_size", (21, 252))
+def test_rolling_window_drop(window_size: int):
+    df = _make_data(n_samples=1_000)
+    rng = np.random.default_rng(0)
+
+    def insert_nulls(val):
+        return None if rng.random() < 0.1 else val
+
+    df = df.with_columns(
+        *(pl.col(c).map_elements(insert_nulls, return_dtype=pl.Float64) for c in ("y",))
+    ).with_row_index()
+
+    df_drop = df.drop_nulls()
+
+    df_merged = (
+        df_drop.lazy()
+        .select(
+            "index",
+            pl.col("y")
+            .least_squares.rolling_ols("x1", "x2", window_size=window_size, mode="predictions")
+            .alias("predictions_1"),
+        )
+        .join_asof(
+            df.lazy().select(
+                "index",
+                pl.col("y")
+                .least_squares.rolling_ols(
+                    "x1",
+                    "x2",
+                    window_size=window_size,
+                    mode="predictions",
+                    null_policy="drop",
```

Comment:
"drop" is also the default for prediction 1?

Reply:
hi @AdrienDart  - yes exactly the default for rolling_ols is set to "drop" (I just set null_policy="drop" to explicitly show). For consistency, I'll add it on the other call though too

Notice that because nulls have been dropped in the first call - any policy (ignore / skip / drop / zero) will return the same result in the first. However ignore/skip will run a little faster as it avoids the slight overhead of maintaining the deque checking for valid observations (as we know all are valid upfront). But we are talking a few microseconds